lib/auth/keystore/aws_kms.go

```diff
@@ -236,16 +256,23 @@ func (a *awsKMSSigner) Sign(rand io.Reader, digest []byte, opts crypto.SignerOpt
 	var signingAlg string
 	switch opts.HashFunc() {
 	case crypto.SHA256:
-		signingAlg = "RSASSA_PKCS1_V1_5_SHA_256"
+		switch a.pub.(type) {
+		case *rsa.PublicKey:
+			signingAlg = kms.SigningAlgorithmSpecRsassaPkcs1V15Sha256
+		case *ecdsa.PublicKey:
+			signingAlg = kms.SigningAlgorithmSpecEcdsaSha256
+		default:
+			return nil, trace.BadParameter("unsupported hash func %q for AWS KMS key type %T", opts.HashFunc(), a.pub)
+		}
 	case crypto.SHA512:
-		signingAlg = "RSASSA_PKCS1_V1_5_SHA_512"
+		signingAlg = kms.SigningAlgorithmSpecRsassaPkcs1V15Sha512
```

Comment:
Yes - are we going to be updating lib/cloud.Clients?

Reply:
yes, but not sure when it will be fully switched. I started on MongoDB IAM auth as I need it for DocumentDB integration and it's simple as it only needs an AWS session. But I run into a few issues:

https://github.com/gravitational/teleport/pull/43497

i will try to put this up for review this week.

Reply:
Looks like your scaffolding a way to add v2 clients to lib/clients there, once the PR is up for review go ahead and ping me and I'll be happy to add kms clients and update this package to use them